Simulate integer index fix

ltsaprounis opened the pull request:
- [X] closes #8487
- [ ] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

The PR is fixing the issue and not breaking any tests in statsmodels/tsa/statespace/tests/test_simulate.py. 
I will also add a test for this edge case before making this ready for review.

</details>

ltsaprounis commented:
@ChadFulton - Thanks for the feedback! I changed the code based on your recommendation in https://github.com/statsmodels/statsmodels/issues/8487 and added a warning and a test.

ChadFulton commented on statsmodels/tsa/statespace/tests/test_simulate.py:
```diff
@@ -1626,6 +1626,28 @@ def test_pandas_univariate_dateindex_repetitions():
+
+    # Fit the forecaster and ensure that the right warning is raised.
+    with pytest.warns(ValueWarning, match=warning_text):
+        forecaster = sarimax.SARIMAX(endog=y, k_factors=1, factor_order=1)
```
`k_factors` and `factor_order` aren't arguments of `SARIMAX`.  I think you want `order`?

ltsaprounis replied:
haha yes - I think I was using UnobservedComponents or DynamicFactor in the notebook I was testing things.
fixed now.

ChadFulton commented on statsmodels/tsa/statespace/tests/test_simulate.py:
```diff
@@ -1626,6 +1626,28 @@ def test_pandas_univariate_dateindex_repetitions():
+    # Fit the forecaster and ensure that the right warning is raised.
+    with pytest.warns(ValueWarning, match=warning_text):
+        forecaster = sarimax.SARIMAX(endog=y, k_factors=1, factor_order=1)
+        forecaster = forecaster.fit()
```
Using `fit` can be problematic in tests (due to numerical instability across platforms or due to the time it takes to perform the fitting), so it's better to just use a fixed set of parameters, e.g. something like:

```python
forecaster = sarimax.SARIMAX(endog=y, order=(1, 0, 0))
forecaster = forecaster.smooth([0.5, 1.0])
```

ChadFulton commented on statsmodels/tsa/base/tsa_model.py:
```diff
@@ -638,6 +638,24 @@ def _init_dates(self, dates=None, freq=None):
+            _index = RangeIndex(index[0], index[-1] + 1)
+            if _index.equals(index):
+                index = _index
+                _index_generated = True
```
Due to the way the `_index_generated` flag is used in the rest of the codebase, we should keep it as `False` even though we are replacing the integer Index with a `RangeIndex` here.

ChadFulton reviewed (commented):
Thanks for updating this PR! It looks good to me.  I just left a couple of easy comments.

ltsaprounis commented:
@ChadFulton - Thanks for the review!

 I have addressed the feedback now. One point that I'm not sure if you would agree with is that I now split the unsupported (but valid) indexes into 2 lists where one is the integer index (with step 1) not starting from zero to allow for testing the different warning messages and `index_generated` flags. The other option I can think of is to add this type of index into the supported ones now.

There's one failing test in the CI: [statsmodels/tsa/statespace/tests/test_dynamic_factor.py::TestDynamicFactor_ar2_errors::test_mle](https://dev.azure.com/statsmodels/statsmodels-testing/_build/results?buildId=4826&view=logs&j=3f9beef4-2a87-5b9a-d136-eeaf4b842580&t=bde20e0b-dfe9-50c5-4f8a-b07f464aecfb&l=399)
The test is failing on a tolerance issue that seems unrelated to this PR, but I'm not sure!
![image](https://user-images.githubusercontent.com/64217214/202902388-93e181a7-569b-4520-9878-f794eeefc727.png)

bashtage commented:
That failure is pretty common on CI/Python 3.9 now.  Need to fix before release, or skip.
